rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml

```diff
@@ -0,0 +1,282 @@
+                    <!-- xslt description="JBoss Web EAP5 (Windup-Generated)" template="transformations/xslt/weblogic-jboss5-web-xml.xsl" extension="-jboss-web.xml"/-->
+                </iteration>
+            </perform>
+        </rule>
+        <!-- TODO this rule is unclear -->
+        <rule id="weblogic-xml-descriptor-13000">
+            <when>
+                <xmlfile as="default" public-id="BEA.+RMI Runtime DTD 1..">
+                </xmlfile>
+            </when>
+            <perform>
+                <iteration>
+                    <classification title="WebLogic RMI XML Version 1.x" severity="mandatory" effort="0"/>
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="weblogic-xml-descriptor-14000">
+            <when>
+                <xmlfile as="default" matches="/wsdd:type-mapping">
+                    <namespace prefix="wsdd" uri="http://www.bea.com/servers/wls70"/>
+                </xmlfile>
+            </when>
+            <perform>
+                <iteration>
+                    <classification title="WebLogic Webservice Type Mapping" severity="mandatory" effort="0"/>
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="weblogic-xml-descriptor-15000">
+            <when>
+                <xmlfile as="default" matches="/wlw:wlw-config">
+                    <namespace prefix="wlw" uri="http://www.bea.com/2003/03/wlw/config/"/>
+                </xmlfile>
+            </when>
+            <perform>
+                <iteration>
+                    <classification title="WebLogic Services Configuration" description="WebLogic specific configuration of web services runtime parameters" severity="mandatory" effort="0"/>
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="weblogic-xml-descriptor-16000">
+            <when>
+                <xmlfile as="default" matches="/project/taskdef[@classname='weblogic.ant.taskdefs.webservices.servicegen.ServiceGenTask'] | /project/taskdef[@classname='weblogic.ant.taskdefs.webservices.javaschema.JavaSchema'] | /project/taskdef[@classname='weblogic.ant.taskdefs.webservices.autotype.JavaSource2DD'] | /project/taskdef[@classname='weblogic.ant.taskdefs.webservices.clientgen.ClientGenTask']">
+                </xmlfile>
+            </when>
+            <perform>
+                <iteration>
+                    <classification title="WebLogic-specific Webservice Ant Tasks" severity="optional" effort="13"/>                    
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="weblogic-xml-descriptor-17000">
+            <when>
+                <xmlfile as="default" matches="/wl:annotation-manifest | /annotation-manifest">
+                    <namespace prefix="wl" uri="http://www.bea.com/2004/03/wlw/external-config/"/>
+                </xmlfile>
+            </when>
+            <perform>
+                <iteration>
```

Comment:
Iteration elements are not required unless you are filtering, selecting variables, or doing some iterating some not iterating conditions in the <perform> clause.